### ISSUE TITLE
Cluster and queue lock

### ISSUE THREAD

**Issue by atoi** [artwells/meteor-queue#8]
In case when application is run by several meteor instances which talk to the same database we have a case when one node obtains a lock and all other nodes just complain to queuelog and do nothing (except making queuelog collection bigger)

Also there is a case when node is shutdown during queue execution and lock is never released.  And no one can obtain it later. You might suggest using purgeOldLocks on app start, but:
1. It would not work as it uses "lockname: {$exists: true}" selector, while the correct name is "lock_name"
2. In a cluster environment we cannot guarantee that we are not killing a lock which is actually used by someone

I think that the cases above could avoided by introducing a way of defining multiple queues, which are run independently. This also could be useful in a single server environment.

**Comment by artwells** [comment]
Thanks for the bug report and the suggestion.

I'll 
1. Fix the purgeOldLocks typo
2. Come up with multiple server environment.  This may just be more options on purgeOldLocks, perhaps reserving tasks for each individual members of a cluster.  In this case, we could suppress logging from other servers at the very least.

Does it sound like I'm understanding you correctly?

**Comment by atoi** [comment]
Yes, I think more options for starting a queue and purging locks should be fine.
Something like:

```
Queue.purgeOldLocks({ownerId: random_string})
Queue.run({ownerId: random_string})
```

But please consider the following scenario (in a single server environment):
```
Queue.add({command: 'task#1, independant'});
Queue.add({command: 'task#2, depends on task#1 results'});
Queue.add({command: 'task#3, independant'});
Queue.add({command: 'task#4, depends on task#3 results'});
```
Currently tasks #3 and #4 will wait till #1 and #2 are finished.

If we had something like this:
```
Queue.add({command: 'some command', queueId: 'someId'});
```
it would be really great. And could be useful in multi-server environment too.

**Comment by artwells** [comment]
the typo issue with purgeOldLocks is fixed in 0.1.1.

I hope to work on the clustering issues soon
